docs(bitbucket-pipelines): authenticated registries (#21096)
renovatebot/renovate: lib/modules/manager/bitbucket-pipelines/readme.md

Extracts dependencies from Bitbucket Pipelines config files.

If you need to change the versioning format, read the [versioning](https://docs.renovatebot.com/modules/versioning/) documentation to learn more.

### Using private build images

Bitbucket Pipelines supports using an authenticated Docker registry, including self-hosted private registries, Docker Hub, Amazon ECR and Google GCR.

> More details can be found on the Atlassian [Bitbucket Pipeline documentation](https://support.atlassian.com/bitbucket-cloud/docs/use-docker-images-as-build-environments) site

Reviewer: ```suggestion
More details can be found on the Atlassian [Bitbucket Pipeline documentation](https://support.atlassian.com/bitbucket-cloud/docs/use-docker-images-as-build-environments) site.
```

Author: Thanks @HonkingGoose 

I'll be updating this PR one @viceice can land https://github.com/renovatebot/renovate/pull/21102

> The following `regexManager` configuration can be used for authenticated Docker Hub or private registries. Changes will be required for Amazon ECR or Google GCR use cases.

Reviewer: ```suggestion
The following `regexManager` configuration can be used for authenticated Docker Hub or private registries.
You'll need to change this example config if you're using Amazon ECR or Google GCR.
```

Also: do our readers know _what_ changes they'll need to make if they're using Amazon ECR or Google GCR? I don't see this explained in the text, but maybe we can assume our readers are competent with ECR/GCR already?

```json
{
  "regexManagers": [
    {
      "matchManagers": ["bitbucket-pipelines"],
      "matchStrings": [
        "\\s*-?\\s?image:\\s*\\n\\s+name:\\s+\"?(?<depName>[a-z/.-]+)(?::(?<currentValue>[a-z0-9.-]+))?(?:@(?<currentDigest>sha256:[a-f0-9]+))?"
      ],
      "datasourceTemplate": "docker",
      "versioningTemplate": "docker"
    }
  ]
}
```

### Replacing unauthenticated with authenticated registries

With the [regex manager](https://docs.renovatebot.com/modules/manager/regex/) you can configure Renovate to assist with migrating your Bitbucket Pipelines from unauthenticated registries to authenticated registries.

> An example configuration is shown below

Reviewer: ```suggestion
For example:
```
